nomad/job_endpoint.go

```diff
@@ -1816,9 +1863,29 @@ func (j *Job) Plan(args *structs.JobPlanRequest, reply *structs.JobPlanResponse)
 
 	// Get the original job
 	ws := memdb.NewWatchSet()
-	existingJob, err := snap.JobByID(ws, args.RequestNamespace(), args.Job.ID)
-	if err != nil {
-		return err
+	var existingJob *structs.Job
+
+	if args.DiffVersion != nil {
+		existingJob, err = snap.JobByIDAndVersion(ws, args.RequestNamespace(), args.Job.ID, *args.DiffVersion)
+		if err != nil {
+			return err
+		}
+		if existingJob == nil {
+			return fmt.Errorf("version %q not found", *args.DiffVersion)
```

Comment:
Note: getting byte output here instead of a version number. I'm doing a bad conversion someplace.

```
nomad job plan -diff-version=7 ../nomad-job-specs/difftester5.nomad
Error during plan: Unexpected response code: 500 (version '\a' not found)

nomad job plan -diff-version=8 ../nomad-job-specs/difftester5.nomad
Error during plan: Unexpected response code: 500 (version '\b' not found)

nomad job plan -diff-version=9 ../nomad-job-specs/difftester5.nomad
Error during plan: Unexpected response code: 500 (version '\t' not found)
```

Reply:
Yeah, that's just how `%q` works. You'll want to use %d there.